src/main/java/orm/assosiation/RelationField.java

```diff
@@ -0,0 +1,109 @@
+package orm.assosiation;
+
+import jakarta.persistence.FetchType;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import orm.TableEntity;
+import orm.settings.JpaSettings;
+import orm.util.ReflectionUtils;
+
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.function.Predicate.not;
+import static orm.util.ReflectionUtils.extractGenericSignature;
+
+public class RelationField {
+
+    // 연관관계 테이블정보
+    private final TableEntity<?> tableEntity;
+
+    // 연관관계 테이블의 조인컬럼명
+    private final String joinColumnName;
+
+    // 연관관계 필드의 값
+    private final Object value;
+
+    private final FetchType fetchType;
+
+    private final boolean isValueTypeCollection;
+
+    private RelationField(Object value, boolean isValueTypeCollection, FetchType fetchType, TableEntity<?> tableEntity, String joinColumnName) {
+        this.value = value;
+        this.isValueTypeCollection = isValueTypeCollection;
+        this.fetchType = fetchType;
+        this.joinColumnName = joinColumnName;
+        this.tableEntity = tableEntity;
+    }
+
+    public RelationField(RelationField relationField) {
+        this.value = relationField.value;
+        this.isValueTypeCollection = relationField.isValueTypeCollection;
+        this.fetchType = relationField.fetchType;
+        this.joinColumnName = relationField.joinColumnName;
+        this.tableEntity = deepCopyTableEntity(relationField.tableEntity);
+    }
+
+    public static RelationField ofOneToManyRelation(Object entity, Field field, JpaSettings settings) {
+        final var tableEntity = new TableEntity<>(extractGenericSignature(field), settings).addAliasIfNotAssigned();
+        final var idField = tableEntity.getId();
+
+        final OneToMany oneToMany = field.getAnnotation(OneToMany.class);
+        final String joinColumnName = Optional.ofNullable(field.getAnnotation(JoinColumn.class).name())
+                .filter(not(String::isBlank))
+                .orElse(idField.getFieldName());
```

Comment:
`@JoinColumn(name = "")` 자체가 없어지면 NPE 가 나더라구요! 이런 모양이 되어야 하지 않을까요?
```suggestion
        final String joinColumnName = Optional.ofNullable(field.getAnnotation(JoinColumn.class))
                .map(JoinColumn::name)
                .filter(not(String::isBlank))
                .orElse(idField.getFieldName());
```